Makefile

```diff
@@ -23,8 +23,37 @@ v8/sd-build.sh: sd-branch
 	@echo "\n\n-------- soft-dev specific info --------"
 	@echo "\nTo build d8 enter the 'v8' dir and run $@"
 
+define RUN_SCRIPT =
+#!/bin/sh
+NPEXECS=30  # Number of process executions (30 in the warmup paper).
+CSV_FILES=sd-v8-results  # File stem only.
+PDF=sd-v8-plots.pdf
+
+# Generate one CSV file for each process execution.
+i=0
+while [ "$i" -lt "$NPEXECS" ]
+do
+    echo Process execution: "$i"
+    tools/sd_run_perf.py --arch x64 --binary-override-path out.gn/x64.release/d8 benchmarks/sd-v8.json --csv-test-results "$CSV_FILES"-"$i".csv
+    i=`expr $i + 1 `
+done
+
+tools/sd_merge_csv.py --files "$CSV_FILES"
+
+../warmup_stats/bin/csv_to_krun_json -l javascript -v V8 -u "`uname -a`" "$CSV_FILES".csv
+
+../warmup_stats --output-plots "$PDF" -l javascript -v V8 -u "`uname -a`" "$CSV_FILES".csv
+endef
+# Export as shell variable.
+export RUN_SCRIPT
+
 v8/sd-run-bench.sh: sd-branch
-	@echo "#!/bin/sh" > $@
-	@echo "tools/run_perf.py --arch x64 --binary-override-path out.gn/x64.release/d8 --buildbot test/js-perf-test/RegExp.json" >> $@
+	@echo "$$RUN_SCRIPT" > $@
 	@chmod +x $@
-	@echo "\nTo run the example benchmark, enter the 'v8' dir and run $@"
+	@echo "To run the example benchmark, enter the 'v8' dir and run $@"
+
+warmup-checkout:
```

Comment:
If you make the name of the target the filename (or in this case dirname) that will be created, then make will know not to try and re-run the target.

Currently, if make was already run, then running `make warmup-checkout` a second time, it will try again since there is nothing on the filesystem named `warmup-checkout`, and this will fail:
```
vext01@bencher8:~/research/sd-fetch-v8$ make warmup-checkout
git clone https://github.com/softdevteam/warmup_stats.git
fatal: destination path 'warmup_stats' already exists and is not an empty directory.
Makefile:56: recipe for target 'warmup-checkout' failed
make: *** [warmup-checkout] Error 128
```

Do you see what I mean?

Reply:
```diff
diff --git a/Makefile b/Makefile
index 5fd463b..a416fff 100644
--- a/Makefile
+++ b/Makefile
@@ -52,8 +52,8 @@ v8/sd-run-bench.sh: sd-branch
        @chmod +x $@
        @echo "To run the example benchmark, enter the 'v8' dir and run $@"

-warmup-checkout:
+warmup_stats:
        git clone https://github.com/softdevteam/warmup_stats.git

-warmup: warmup-checkout
+warmup: warmup_stats
        cd warmup_stats && ./build.sh
```

```
vext01@bencher8:~/research/sd-fetch-v8$ file warmup_stats
warmup_stats: directory
vext01@bencher8:~/research/sd-fetch-v8$ make warmup_stats
make: 'warmup_stats' is up to date.
```